src/pages/workspaces/workspace/analysis/modals/ComputeModal.js

```diff
@@ -1626,7 +1624,16 @@ export const ComputeModalBase = ({
           hideCloseButton: shouldHideCloseButton,
           onDismiss
         }),
-        div(['A cloud environment consists of application configuration, cloud compute and persistent disk(s).'])
+        div(['A cloud environment consists of application configuration, cloud compute and persistent disk(s).']),
+        h(IdContainer, [
```

Comment:
Since there's only one compute modal on screen at a time, do we need a generated unique ID here? Or can it be hard coded, as above with `h(TitleBar, { id: titleId ...`?

Or, could the compute modal have one unique ID and derive other element IDs from that?
```js
const modalId = useUniqueId()
...
h(TitleBar, { id: `${modalId}-some-section-title` ...
```

Reply:
I'm going to use a string here, because the useUniqueId would require me converting this to a react hook, which I feel would be better in another ticket, so as to prevent more scope creep

Reply:
👍 I don't think it would require converting anything to a hook. ComputeModal could call `useUniqueId` at the top of the function where it calls other hooks and the ID could be used throughout. But since there won't be multiple instances of compute modal on screen at once, a hard coded string works here too.

Reply:
alright I'll keep it hardcoded, but as it looks like I may be the person who ends up converting computemodal to TS, I'll keep it in mind to use useUniqueId with it later